AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/editor_component/editor_component_validation.py

```diff
@@ -0,0 +1,210 @@
+    y_is_close = Math.Math_IsClose(expected.get_property('y'), actual.get_property('y'), 0.001)
+    z_is_close = Math.Math_IsClose(expected.get_property('z'), actual.get_property('z'), 0.001)
+
+    return x_is_close and y_is_close and z_is_close
+
+def _validate_xyz_is_float(x: float, y: float, z: float, error_message: str) -> None:
+    """
+    Helper Function for Editor Components to test if passed in X, Y, Z used in editor component tests are
+       all actually floats.
+
+    Note: This function is not intended to be used at test execution time, and is structured as such.
+
+    x: The x-point position to validate if it's a float.
+    y: The y-point position to validate if it's a float.
+    z: The z-point position to validate if it's a float.
+    error_message: The error message to assert if values are not float.
+
+    return: boolean result of testing whether each input is a float.
+    """
+    assert isinstance(x, float) and isinstance(y, float) and isinstance(z, float), error_message
+
+def _validate_property_visibility(component: EditorComponent, component_property_path: str, expected: str) -> None:
+    """
+    Helper function for Editor Components to validate the current visibility of the property is expected value.
+
+    Valid values for Property Visibility found in EditorPythonTestTools.consts.general ComponentPropertyVisibilityStates
+    """
+    Report.info(f"Validating visibility for {component.get_component_name()} "
+                f"component at property path {component_property_path} is \"{expected}\"")
+    assert expected in vars(PropertyVisibility).values(), \
+        f"Expected value of {expected} was not an expected visibility state of: {vars(PropertyVisibility).values()}"
+
+    visibility = component.get_property_visibility(component_property_path)
+    assert visibility == expected, \
+        f"Error: {component.get_component_name()}'s component property visibility found at {component_property_path} " \
+        f"was set to \"{visibility}\" when \"{expected}\" was expected."
+
+def validate_integer_property(property_name: str, get_int_value: typing.Callable, set_int_value: typing.Callable,
+                              component_name: str, tests: typing.Dict[str, typing.Tuple[int, bool]]) -> None:
+    """
+    Function to validate the behavior of a property that gets and sets an Integer value.
+
+    :get_float_value: The Editor Entity Component Property's get method.
+    :set_float_value: The Editor Entity Component Property's set method.
+    :component_name: The name of the Editor Entity Component under test.
+    :property_name: The name of the Editor Entity Component Property under test.
+    :tests: A dictionary that stores an Int value and the intent of the values being passed. The following
+    format is expected
+     {
+        "test_description": (int_value, expect_pass),
+        "Zero value Test": (0, True)
+     }
+    """
+    int_value, expect_pass = 0, 1
```

Comment:
I can change this as I normally do this in other languages. However, this is consider a "pythonic style" in a conventional styling break from most languages that support this, since behind the scenes this is considered "tuple unpacking". Due to this I decided to use it here.

That said, the audience for reading these scripts are not necessarily "python programmers", so I can use the more standardized behaviors.

To answer your question, python will treat the left side as individual values to be assigned, and the right side is treated as a tuple to be unpacked, assigning the values in left → right order via an iterator. If there are more values than variables, the last values are not assigned. If there are more variables than values, the remaining variables will be set to `None`.

Another trick that python supports is setting a bunch of variables to the same value using equals operators between them.

EG:
`a = b = c = "Orange"` would set all the variables to the string value "Orange"

Reply:
Now using tuple unpacking.

Reply:
Thanks for the explanation. FYI this is the part of our standards where it talks about this: https://github.com/o3de/sig-core/blob/main/governance/Coding-Standards-and-Style-Guide.md#single-statement-per-line